frontend/src/app/shared/components/op-date-picker/datepicker.ts

```diff
@@ -25,57 +25,53 @@
 //
 // See COPYRIGHT and LICENSE files for more details.
 //++
-
 import * as moment from 'moment';
 import flatpickr from 'flatpickr';
-import { Instance } from 'flatpickr/dist/types/instance';
+import {
+  DayElement,
+  Instance,
+} from 'flatpickr/dist/types/instance';
 import { ConfigurationService } from 'core-app/core/config/configuration.service';
 import { I18nService } from 'core-app/core/i18n/i18n.service';
 import { rangeSeparator } from 'core-app/shared/components/op-date-picker/op-range-date-picker/op-range-date-picker.component';
+import { WeekdayResourceService } from 'core-app/core/state/days/weekday.service';
+import { IWeekday } from 'core-app/core/state/days/weekday.model';
+import { Injector } from '@angular/core';
+import { InjectField } from 'core-app/shared/helpers/angular/inject-field.decorator';
+import { take } from 'rxjs/operators';
 import DateOption = flatpickr.Options.DateOption;
 
 export class DatePicker {
   private datepickerFormat = 'Y-m-d';
 
-  private datepickerCont:HTMLElement = document.querySelector(this.datepickerElemIdentifier)!;
+  private datepickerCont:HTMLElement = document.querySelector(this.datepickerElemIdentifier) as HTMLElement;
 
   public datepickerInstance:Instance;
 
-  private reshowTimeout:any;
+  private reshowTimeout:ReturnType<typeof setTimeout>;
+
+  private weekdays:IWeekday[] = [];
+
+  @InjectField() configurationService:ConfigurationService;
 
-  constructor(private datepickerElemIdentifier:string,
+  @InjectField() weekdaysService:WeekdayResourceService;
+
+  @InjectField() I18n:I18nService;
```

Comment:
Because the Datepicker class is not a service, but a standard typescript class called / instantiated in other places. You can of course pass all dependencies directly, but that would extend the constructor call, and that's what we have the injector for in angular.